server/src/test/java/io/crate/expression/scalar/ArrayAppendFunctionTest.java

```diff
@@ -0,0 +1,205 @@
+    }
+
+    @Test
+    public void test2DArrayArgumentWithArrayObjectIncludingNullValue() throws Exception {
+        assertNormalize("array_append([[1,2],[3,4]], [null, 6])", isLiteral(Arrays.asList(Arrays.asList(1,2), Arrays.asList(3,4), Arrays.asList(null,6))));
+    }
+
+    @Test
+    public void testEmptyArrayNullObject() throws Exception {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("The array argument of the `array_append` function can be of " +
+                                        "undefined type or of undefined inner type, but not when object argument " +
+                                        "is also undefined");
+        assertNormalize("array_append([], null)", null);
+    }
+
+    @Test
+    public void testNullArrayNullObject() throws Exception {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("The array argument of the `array_append` function can be of " +
+                                        "undefined type or of undefined inner type, but not when object argument " +
+                                        "is also undefined");
+        assertNormalize("array_append(null, null)", null);
+    }
+
+    @Test
+    public void testNullArrayInnerTypeNullObject() throws Exception {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("The array argument of the `array_append` function can be of " +
+                                        "undefined type or of undefined inner type, but not when object argument " +
+                                        "is also undefined");
+        assertNormalize("array_append([null], null)", null);
+    }
+
+    @Test
+    public void testZeroArguments() throws Exception {
+        expectedException.expect(UnsupportedOperationException.class);
+        expectedException.expectMessage("Unknown function: array_append()." +
+                                        " Possible candidates: array_append(array(E), E):array(E)");
+        assertEvaluate("array_append()", null);
+    }
+
+    @Test
+    public void testOneArgument() {
+        expectedException.expect(UnsupportedOperationException.class);
+        expectedException.expectMessage("Unknown function: array_append(_array(1))," +
+                                        " no overload found for matching argument types: (integer_array).");
+        assertEvaluate("array_append([1])", null);
+    }
+
+    @Test
+    public void testThreeArguments() throws Exception {
+        expectedException.expect(UnsupportedOperationException.class);
+        expectedException.expectMessage("Unknown function: array_append(_array(1), _array(2), _array(3))," +
+                                        " no overload found for matching argument types: (integer_array, integer_array, integer_array).");
+        assertEvaluate("array_append([1], [2], [3])", null);
+    }
```

Comment:
This is testing our general function matching logic and is already tested at the related components. I'd suggest to just remove these here.

Reply:
Removed :)